axelrod/match.py

```diff
@@ -77,11 +82,28 @@ def play(self):
 
         i.e. One entry per turn containing a pair of actions.
         """
+        if self._prob_end is not None:
```

Comment:
Well as I go up through the classes I'll keep pushing to this branch. It might well be that it's not quite right just yet...